wiki/DEBUGGING.md

```diff
@@ -0,0 +1,128 @@
+Courseography is packed with Monads!
+
+### Example
+
+First we add `traceM` after the monad we want to view:
+
+![Image](images/debugTrace.png)
+
+Then after running the program in the terminal, we get to see our monad's value!:
+
+![Image](images/debugTraceOutput.png)
+
+## GHCi Debugger
+
+### Setup
+
+After setting Courseography as the root in terminal, type in `stack ghci`. This will set up the environment for
+debugger. Note this will not work if your program has compilation errors. Courseography has a lot of files already
+embedded in a package, which stack automatically recognizes. If there is a case where a module `module0` is not imported,
+you can load in `module0` by typing `:l module0` into the GHCi environment. This command can also load multiple files
+at the same time: `:l module1 module2 module3 ...`. One more thing to note is that dependencies have to be imported at
+the same time, or in order of dependency; meaning that if I am importing `module4` in `module5`, I have to import
+both at the same time, or `module4` first.
+
+### Breakpoints
+
+Setting up breakpoints is done by the command `:break ModuleA.my_func` in the case you want to set the breakpoint at
+a specific function in the module or `:break ModuleA line_number` if you want to set the breakpoint at a specific line.
+In order to see the breakpoints set, use the command `:show breaks`. Breakpoints can be enabled/disabled with the
+command `:enable/:disable breakpoint_number`. To enable/disable all breakpoints at the same time use
+`:enable/:disable *`. Finally, to delete breakpoints use the command `:delete breakpoint_number` or to delete all use
+`:delete *`.
+
+### Program Trace
+
+Once you have your breakpoints set, you can start running the program through the main entrypoint `main`. After, the
+debugger will stop wherever you set your breakpoint. From this point the relevant variables and functions the program
+is using should be viewable. Use the command `:info var_name` to see the object's type and `:print var_name` to see its
+value. **Note that you can only use the main entry point once, i.e. you will have to restart the debugger**.
+
+Stepping through the program can be done using `:step`. This will step through the program until the next breakpoint.
+There are two variants `:steplocal` and `:stepmodule`, where the former limits the breakpoints that are enabled to the
+scope of the current function and the latter restricts the scope of the breakpoints to the current module. **To reiterate
+the point about the main entry point in the last paragraph, the first time `main` is called it will trace step by step
+until the result of the entire program is printed, but if main is called again in the same session, it will be evaluated
+and the stepper environment will not be available**.
+
+Functions can also be called. Type the function and its input parameters directly into the debugger: `function inputs`.
```

Comment:
You don't need to provide an example, this is standard function call syntax. You just need to say that breakpoints can apply to functions called in ghci directly.